sdk/core/core/inc/az_span.h

```diff
@@ -135,161 +158,306 @@ AZ_NODISCARD az_span az_span_from_str(char* str);
- * span using case-insensitive compare.
+ * @brief az_span_is_content_equal_ignoring_case returns `true` if the lengths and characters
+ * referred to by \p span1 and \p span2 are identical except for case. This function assumes the bytes in
+ * both spans are ASCII characters.
  *
- * @param a comparing span
- * @param b comparing span
- * @return AZ_NODISCARD az_span_is_equal_content_ignoring_case
+ * @return Returns true if the lengths of both spans are identical and the ASCII characters in both
+ * spans are also identical except for case.
  */
-AZ_NODISCARD bool az_span_is_content_equal_ignoring_case(az_span a, az_span b);
+AZ_NODISCARD bool az_span_is_content_equal_ignoring_case(az_span span1, az_span span2);
 
 /**
- * @brief convert all the span content to uint64
+ * @brief az_span_to_str copies a source span containing a string (not 0-terminated) to a
+ destination char buffer and appends the 0-terminating byte.
+
+ * The buffer referred to by destination must have a max_size that is at least 1 byte bigger
+ than the source az_span
+ *  the string in converts \p destination span to zero-terminated str. Content is copied to \p source
+ buffer and then \0 is
+ * addeed at the end. Then out_result will be created out of buffer
  *
- * @param self source span
- * @param out result span
- * @return AZ_NODISCARD az_span_to_uint64
+ * @param[in] destination A pointer to a buffer where the string should be copied
+ * @param[in] destination_max_size The maximum available space within the buffer referred to by destination.
+ * @param[in] source The az_span containing the not-0-terminated string
+ * @return An #az_result value indicating the result of the operation.
+ *          #AZ_OK If /p source span's content is successfully copied to the destination.
+ *          #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY if the /p destination buffer is too small to
+ copy the string and 0-terminate it
  */
-AZ_NODISCARD az_result az_span_to_uint64(az_span self, uint64_t* out);
+AZ_NODISCARD az_result
+az_span_to_str(char* destination, int32_t destination_max_size, az_span source);
+
+/******************************  SPAN PARSING */
 
 /**
- * @brief convert all the span content to uint32
+ * @brief az_span_to_uint64 parses an az_span containing ASCII digits into a uint64 number
  *
- * @param self source span
- * @param out result span
- * @return AZ_NODISCARD az_span_to_uint32
+ * @param[in] span The az_span containing the ASCII digits to be parsed.
+ * @param[in] out_number The pointer to the variable that is to receive the number
+ * @return An #az_result value indicating the result of the operation.
+ *          #AZ_OK if successful
+ *          #AZ_ERROR_PARSER_UNEXPECTED_CHAR if a non-ASCII digit is found within the span.
  */
-AZ_NODISCARD az_result az_span_to_uint32(az_span self, uint32_t* out);
+
+AZ_NODISCARD az_result az_span_to_uint64(az_span span, uint64_t* out_number);
```

Comment:
Yes, I like source

Reply:
@JeffreyRichter This is a large change.  We use parameters named "span" in a lot of places.  Also that might mean we want other places to be source1 source2
`AZ_NODISCARD AZ_INLINE bool az_span_is_content_equal(az_span span1, az_span span2)`

What about out_span?  I don't have a better way to describe an appended span.  Thus we will still have span on some parameters i.e. out_span

Reply:
That's fine, we can leave it as is.

Reply:
I think its fine to keep using `span1`, `span2` where it makes sense and same with `out_span`. For the examples you shared, those parameter names have useful meaning. I am not sure that just because we have use of `span1`, `span2`, `out_span`, we should leave `span` as is. To me, those points are a bit unrelated. `span1`/`span2` isn't necessarily redundant, `span` is. In the places where `source` is more meaningful (when we have a single source span), we should use it. That said, I don't feel too strongly about it, and we can leave it as is if that's the consensus :)